prov/efa/src/efa_mr.c

```diff
@@ -261,7 +261,12 @@ static int efa_mr_hmem_setup(struct efa_mr *efa_mr,
 				EFA_WARN(FI_LOG_MR,
 					 "Unable to register handle for GPU memory. err: %d buf: %p len: %zu\n",
 					 err, attr->mr_iov->iov_base, attr->mr_iov->iov_len);
-				return err;
+				/* When gdrcopy pin buf failed, fallback to cudaMemcpy when user enables cuda xfer */
+				if (efa_mr->domain->cuda_xfer_setting != CUDA_XFER_ENABLED)
+					return err;
+				efa_mr->peer.device.cuda = attr->device.cuda;
```

Comment:
Missing `efa_mr->peer.use_gdrcopy = false` - could cause uninitialized variable issue.

Reply:
I thought about that. I could add it explicitly, but I assume this should be handled by the very beginning like `efa_mr = {0}`

Reply:
This is unnecessary in new revision as we will use `device.reserved` which is reset to 0 already.

Reply:
The latest revision set efa_mr->peer.use_gdrcopy = false at the beginning.